### PR TITLE
Fix TypeError in Reporter Web App

### PR DESCRIPTION
**Pull request by tsteenbe**
See individual commits for details - this PR should fix most issues but found some more edge case bugs but those I fill fix in follow-up PR.

### CODE REVIEW

File: reporter-web-app/config/webpack.config.js [hunk 1]
```diff
@@ -88,23 +87,24 @@ module.exports = function (webpackEnv) {
...
-              },
-              stage: 3,
-            }),
-            // Adds PostCSS Normalize as the reset css with default options,
-            // so that it honors browserslist config in package.json
-            // which in turn let's users customize the target behavior as per their needs.
-            postcssNormalize(),
-          ],
-          sourceMap: isEnvProduction && shouldUseSourceMap,
+          postcssOptions: {
```
Comment (fviernau) [inline reporter-web-app/config/webpack.config.js:90]: commit-msg: We usually try to wrap at 72 chars.

File: reporter-web-app/src/components/PackageDetails.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ const PackageDetails = (props) => {
                 {id}
             </Item>
             {
-                purl.length !== 0
+                !!purl
```
Comment (fviernau) [inline reporter-web-app/src/components/PackageDetails.js:63]: Can you give me a hint about the semantics of `!!`?

In particular also, is the length check still covered?

File: reporter-web-app/src/models/WebAppPackage.js [hunk 3]
```diff
@@ -86,7 +86,7 @@ class WebAppPackage {
 
     #projectIndexes;
 
-    #purl;
+    #purl = '';
```
Comment (fviernau) [inline reporter-web-app/src/models/WebAppPackage.js:89]: Why is this needed?

File: reporter-web-app/src/components/IssuesTable.js [hunk 4]
```diff
@@ -262,36 +262,36 @@ const IssuesTable = (props) => {
                             {
                                 webAppOrtIssue.isResolved
                                 && (
-                                    <Panel header="Resolutions" key="0">
+                                    <Panel header="Resolutions" key="1">
```
Comment (fviernau) [inline reporter-web-app/src/components/IssuesTable.js:265]: Commit message: Alings -> Align

File: reporter-web-app/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
-    "name": "@here/oss-review-toolkit-reporter",
-    "description": "React-based App to visualize scan reports generated by OSS Review Toolkit",
+    "name": "ort-web-app-reporter",
```
Comment (fviernau) [inline reporter-web-app/package.json:2]: commit message: "Rename package name" -> how about just "Rename the package"

**Review by fviernau — CHANGES_REQUESTED** [review]

File: reporter-web-app/src/components/PackageDetails.js (hunk 2 above)
Comment (tsteenbe) [reply reporter-web-app/src/components/PackageDetails.js:63]: ! forces a boolean from any type and !! inverts the first evaluated boolean e.g. not(true) / not (false). The original condition purl.length !== 0 check if purl string was greater than zero but did not check if purl was undefined that why we got type error can determine length of undefined. Using !! I no check undefined and string length in one statement as in JavaScript boolean(undefined) == false, boolean('') == false.

File: reporter-web-app/src/components/IssuesTable.js (hunk 4 above)
Comment (tsteenbe) [reply reporter-web-app/src/components/IssuesTable.js:265]: Fixed

File: reporter-web-app/package.json (hunk 5 above)
Comment (tsteenbe) [reply reporter-web-app/package.json:2]: Fixed

File: reporter-web-app/src/components/PackageDetails.js (hunk 2 above)
Comment (fviernau) [reply reporter-web-app/src/components/PackageDetails.js:63]: Thanks!

File: reporter-web-app/config/webpack.config.js (hunk 1 above)
Comment (tsteenbe) [reply reporter-web-app/config/webpack.config.js:90]: Fixed

File: reporter-web-app/src/models/WebAppPackage.js (hunk 3 above)
Comment (tsteenbe) [reply reporter-web-app/src/models/WebAppPackage.js]: Oops, was left-over form debugging removed it now

**Review by fviernau — APPROVED** [review]
